src/WebJobs.Script.Grpc/Extensions/RpcSharedMemoryDataExtensions.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Grpc.Messages;
+using Microsoft.Azure.WebJobs.Script.Workers.SharedMemoryDataTransfer;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Grpc.Extensions
+{
+    internal static class RpcSharedMemoryDataExtensions
+    {
+        internal static async Task<RpcSharedMemory> ToRpcSharedMemoryAsync(this object value, ILogger logger, string invocationId, ISharedMemoryManager sharedMemoryManager)
+        {
+            if (!sharedMemoryManager.IsSupported(value))
+            {
+                return null;
+            }
+
+            // Put the content into shared memory and get the name of the shared memory map written to
+            SharedMemoryMetadata putResponse = await sharedMemoryManager.PutObjectAsync(value);
+            if (putResponse == null)
+            {
+                logger.LogTrace("Cannot write to shared memory for invocation id: {Id}", invocationId);
+                return null;
+            }
+
+            // If written to shared memory successfully, add this shared memory map to the list of maps for this invocation
+            sharedMemoryManager.AddSharedMemoryMapForInvocation(invocationId, putResponse.Name);
+
+            RpcDataType? dataType = GetRpcDataType(value);
+            if (!dataType.HasValue)
+            {
+                logger.LogTrace("Cannot get shared memory data type for invocation id: {Id}", invocationId);
+                return null;
+            }
+
+            // Generate a response
+            RpcSharedMemory sharedMem = new RpcSharedMemory()
+            {
+                Name = putResponse.Name,
+                Offset = 0,
+                Count = putResponse.Count,
+                Type = dataType.Value
+            };
+
+            logger.LogDebug("Put object in shared memory for invocation id: {Id}", invocationId);
```

Comment:
Why is this Debug when some of the other related logs are Trace? Note that there is overhead for Debug messages, since they will be logged to Kusto by our system logger (ETW events) - system logger defaults to log level Debug. How many new per invocation Debug or higher level logs end up getting written when shared memory is used?

Reply:
Just for my clarification, if I do `LogTrace` it will *not* show up in Kusto?
The intention here is to track how many objects we were able to send via shared memory after this change goes in. I think this is the only `LogDebug` message when shared memory is used so only N of these will be written where N is the number of binding objects transferred over shared memory per invocation.

Reply:
Please change this so that only one log is emitted per invocation.

Reply:
Changed it so that there is one `LogDebug` at the time of sending the invocation (to log the number of bytes *sent* over shared memory) and one at the time of getting the response (to log the number of bytes *received* over shared memory.)